### DISCUSSION TITLE
Oracle Java SE - October 2021

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#6143] [Unsupported Versions] [answered]
Our Nessus scans is finding an outdated version of Java SE in one of the containers of our SO box.

Here's an excerpt of the results:
The following vulnerable instances of Java are installed on the
remote host :
 Installed version : 1.11.0_11
 Fixed version     : 1.7.0_321 / 1.8.0_311 / 1.11.0_13 / 1.17.0_1

I just ran soup and it still persists.  Can anyone tell me 1.11.0_13 or newer will roll downstream so it will make my scans happy?  Our management gets nervous when any high risk item is picked up on our scans.

Thanks in advance for the help and for such an amazing product.

**Comment by dougburks** [comment, marked as answer]
Does your Nessus scan say which container it found this in? I suspect it is finding the Java included in the upstream Elastic containers. If so, this should already be fixed in the upcoming 2.3.90 release since we have moved to Elastic 7.15.1. 

From https://www.elastic.co/guide/en/elasticsearch/reference/7.15/release-notes-7.15.1.html:

> Upgrade bundled JDK to Java 17 #[78197](https://github.com/elastic/elasticsearch/pull/78197) (issue: #[78168](https://github.com/elastic/elasticsearch/issues/78168))

**Reply by DebianGuru** [reply]
Doug, Thanks for the reply.
This is more of the output from Nessus:

  Path              : /var/lib/docker/overlay2/4fb8c5b1a1486d300b5423fee88bddeed529182e9ba2985e5f94b3b36a5bbe4d/merged/usr/share/logstash/jdk
  Path              : /var/lib/docker/overlay2/577ea4b8b054f83b584fca2e46ebd01a45c5a9d8ea7482bfcf2ffaceeeca6bf0/diff/usr/share/logstash/jdk
  Path              : /var/lib/docker/overlay2/6cbba1d70d2459781740b3316719b0f7c198f62b65d4e7ad75c93e26700da49a/diff/usr/share/logstash/jdk
  Path              : /var/lib/docker/overlay2/d1352b7ba873c142c83ace9306b87997a9bbd817056acafe1ff63be46f834777/diff/usr/share/logstash/jdk
  Installed version : 1.11.0_11
  Fixed version     : 1.7.0_321 / 1.8.0_311 / 1.11.0_13 / 1.17.0_1

It appears to be logstash if I'm interpreting that correctly.

**Reply by dougburks** [reply]
Yes, that does appear to be Logstash. Our upcoming 2.3.90 release will include Logstash 7.15.1 which includes a newer JDK:
https://www.elastic.co/guide/en/logstash/7.15/logstash-7-15-1.html

**Reply by DebianGuru** [reply]
Thanks Doug,

I appreciate your feedback very much.  However, this news isn't as good as I'd wished.

According to that link, it will still only include JDK 11.0.12+7, which falls short of what Nessus wants to mitigate the CVEs listed.  That's if I understand the Java versioning scheme.    The post from the link was from back in August and all the CVEs are from October.  I guess their release cycle is slow.  Does your team have any official contacts with them to see if this is the case? (Don't know if you're partners with them and have an inside track.)

Keeping up with Java updates is like playing  "Whack-a-Mole".  When building a system with Docker, does it put you at the mercy of whoever maintains the other containers to keep everything updated?  (I'm not trying to be critical, I really don't understand.  It is often the case that one thing depends on the other.)

Thanks again for the help.

**Reply by dougburks** [reply]
Elastic 7.15.2 dropped today and so we'll be including that in our upcoming 2.3.90 release. It's possible that they've updated Java again although it's not currently listed in the release notes:
https://www.elastic.co/guide/en/logstash/7.15/logstash-7-15-2.html

Elastic is usually pretty good at keeping up with Java updates, so if it's not in 7.15.2, I would suspect it would be in the next release.

In the meantime, the Nessus plugin that you're referencing appears to be https://www.tenable.com/plugins/nessus/154344. I've pasted 3 paragraphs and bolded and italicized some important notes:

================ BEGIN COPYPASTA ==============================================

- Vulnerability in the Java SE product of Oracle Java SE (component: JavaFX (libxml)). The supported version that is affected is Java SE: 8u301. Easily exploitable vulnerability allows unauthenticated attacker with network access via multiple protocols to compromise Java SE. Successful attacks of this vulnerability can result in unauthorized ability to cause a hang or frequently repeatable crash (complete DOS) of Java SE as well as unauthorized update, insert or delete access to some of Java SE accessible data and unauthorized read access to a subset of Java SE accessible data. **_This vulnerability applies to Java deployments, typically in clients running sandboxed Java Web Start applications or sandboxed Java applets, that load and run untrusted code (e.g., code that comes from the internet) and rely on the Java sandbox for security. This vulnerability does not apply to Java deployments, typically in servers, that load and run only trusted code (e.g., code installed by an administrator). (CVE-2021-3517)_**

- Vulnerability in the Java SE product of Oracle Java SE (component: Deployment). The supported version that is affected is Java SE: 8u301. Difficult to exploit vulnerability allows unauthenticated attacker with network access via multiple protocols to compromise Java SE. Successful attacks require human interaction from a person other than the attacker. Successful attacks of this vulnerability can result in takeover of Java SE. **_This vulnerability applies to Java deployments, typically in clients running sandboxed Java Web Start applications or sandboxed Java applets, that load and run untrusted code (e.g., code that comes from the internet) and rely on the Java sandbox for security. This vulnerability does not apply to Java deployments, typically in servers, that load and run only trusted code (e.g., code installed by an administrator). (CVE-2021-35560)_**

- Vulnerability in the Java SE, Oracle GraalVM Enterprise Edition product of Oracle Java SE (component: Libraries). Supported versions that are affected are Java SE: 8u301, 11.0.12, 17; Oracle GraalVM Enterprise Edition: 20.3.3 and 21.2.0. Easily exploitable vulnerability allows low privileged attacker with network access via Kerberos to compromise Java SE, Oracle GraalVM Enterprise Edition. **_Successful attacks require human interaction from a person other than the attacker_** and while the vulnerability is in Java SE, Oracle GraalVM Enterprise Edition, attacks may significantly impact additional products. Successful attacks of this vulnerability can result in unauthorized access to critical data or complete access to all Java SE, Oracle GraalVM Enterprise Edition accessible data. (CVE-2021-35567)

================ END COPYPASTA ==============================================

The bolded statements should dramatically lower the severity of this finding based on how we actually use Logstash. Also keep in mind compensating controls like the host-based firewall so Logstash is not exposed to the network by default.

**Reply by DebianGuru** [reply]
That's awesome!  The plugin number in my Nessus install which caught that is #154345, but it's the same CVEs that it references. Thanks a million!  I forwarded that to my manager and he's happy with that detailed explanation.
